runtime/src/lib.rs

```diff
@@ -118,6 +118,14 @@ pub trait Clock: Clone + Send + Sync + 'static {
     /// Returns the current time.
     fn current(&self) -> SystemTime;
 
+    /// Returns the UNIX epoch time.
+    /// Panics if the system clock is set to before the Unix epoch.
+    fn epoch(&self) -> Duration {
+        self.current()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("failed to get epoch time")
+    }
+
```

Comment:
is it acceptable to define this in the trait? 😅

Reply:
Makes sense to me. This is considered a "default method" (that can still be implemented) but would be the same in both implementations.

I wonder if we should do the same for `sleep` (which would always calls `sleep_until` with `self.current().checked_add(duration)`)?

Reply:
It'll compile (as you've seen) as a "default method" for the trait (which can be overridden).

I consider this a style discussion. IMO we want to keep traits as minimal as possible to minimize the burden of creating a `runtime` and would prefer we implement this as public wrapper function that takes in some `Clock` trait? I think we could probably do the same for `sleep` (which just calls `sleep_until` under the hood).

Reply:
Would it make more sense to convert `current` to return `Duration`? We may never actually use it directly as `SystemTime`...

Reply:
I would recommend putting this in `runtime/utils` as `epoch<C: Clock>(clock: &C)`

Reply:
Then you could use it as `epoch(&self.runtime)`